Includes/findXBE.cpp

```diff
@@ -21,15 +22,35 @@ int findXBE(void* list) {
       sprintf(tmp, "%s%s\\default.xbe", path, fData.f_FileName);
       tmpFILE = fopen(tmp, "rb");
       if (tmpFILE != nullptr) {
-        xbeMenuItem* itm = new xbeMenuItem(fData.f_FileName, tmp);
-        gmi_list->push_back(*itm);
+        read_bytes = fread(xbeData, 1, SECTOR_SIZE, tmpFILE);
+        XBE *xbe = (XBE*)xbeData;
+        if(xbe->type != XBETYPESTRING ||
+           xbe->baseAddress != BASEADDRESS ||
+           xbe->baseAddress > xbe->certAddress ||
+           xbe->certAddress + 4 >= (xbe->baseAddress + xbe->sizeOfHeaders) ||
+           xbe->sizeOfHeaders > read_bytes) {
+          continue;
+        }
+        XBECert *xbeCert = (XBECert*)&xbeData[xbe->certAddress -
+                                              xbe->baseAddress];
+        int offset = 0;
+        while(offset < XBENAMESIZE) {
+          sprintf(&xbeName[offset], "%c", (char)xbeCert->xbeName[offset]);
+          if(xbeCert->xbeName[offset] == 0x0000) {
+            break;
+          }
+          ++offset;
+        }
+        gmi_list->push_back(xbeMenuItem(xbeName,tmp));
         fclose(tmpFILE);
         tmpFILE = nullptr;
+        memset(xbeName, 0, XBENAMESIZE);
```

Comment:
Garbage in RAM isn't an issue, this should probably be done when it actually needs to be zero (before you start re-using it?). Also consider `memset(xbeName, 0x00, sizeof(xbeName))` to avoid issues when the array is being resized (size changes are more likely than changing it into a pointer).